prototype2/common/Buffer.h

```diff
@@ -0,0 +1,90 @@
+/// Copyright (C) 2016-2018 European Spallation Source, see LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief common definition of buffer, used as reference to some location in
+///        memory with a length. NOT responsible for (de-)allocation!
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <cinttypes>
+#include <cstddef>
+#include <vector>
+
+// \todo reverse iteration facilities
+
+template<typename T>
+struct Buffer {
+
+  // \brief creates invalid buffer
+  Buffer() {}
+
+  // \brief creates buffer with provided address and size
+  Buffer(T *addr, size_t sz)
+      : address(addr), size(sz) {}
+
+  // \brief creates buffer as reference to standard vector
+  Buffer(std::vector<T> &vector)
+      : Buffer(vector.data(), vector.size()) {}
+
+  size_t bytes() const {
+    return size * sizeof(T);
+  }
+
+  // \brief creates buffer from buffer, converting bit resolution
+  template<typename TT>
+  explicit Buffer(Buffer<TT> other)
```

Comment:
This constructor seems dangerous to me as it does a `reinterpret_cast`. Can you give me a good use case for it?

Reply:
There is a test for this, where we convert Buffer resolution. And this is used at least once in Multigrid parsing, where what you get is in char, but of course you want to look at it in uint32